### PR TITLE
Improve doc test coverage(files)

### PR DESCRIPTION
**Pull request by mlopatkin**
Part of #21027

### CODE REVIEW

File: subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/files/SamplesCopyIntegrationTest.groovy [hunk 1]
```diff
@@ -240,6 +241,141 @@ class SamplesCopyIntegrationTest extends AbstractSampleIntegrationTest {
...
+        outputDir.file('index-staging.html').isFile()
+        outputDir.file('products/gradle-staging.html').isFile()
+        outputDir.file('asset1.txt').isFile()
+
+        where:
+        dsl << ['groovy', 'kotlin']
+    }
+
+    @UsesSample("files/copy")
+    @ToBeFixedForConfigurationCache(because = "https://github.com/gradle/gradle/issues/22536")
```
Comment (mlopatkin) [inline subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/files/SamplesCopyIntegrationTest.groovy:304]: My plan is to add a warning similar to what we're going to have in #22683 to the doc and rewrite the snippet as part of #22536.

**Comment by mlopatkin** [comment]
@bot-gradle test PRF

**Comment by mlopatkin** [comment]
@bot-gradle test PRF

**Comment by mlopatkin** [comment]
@bot-gradle test BD, PRF

**Comment by mlopatkin** [comment]
The updated docs section:
- [working_with_files](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/58133052:id/distributions/gradle-8.0-docs.zip!/gradle-8.0-20221114181752%2B0000/docs/userguide/working_with_files.html#sec:project_copy_method)

**Review by 6hundreds — APPROVED** [review]
LGTM!

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**